smart_open/tests/test_smart_open.py

```diff
@@ -1840,6 +1841,176 @@ def test(self):
         self.assertEqual(expected, actual)
 
 
+_RAW_DATA = "не слышны в саду даже шорохи".encode("utf-8")
+
+
+class HandleS3CompressionTestCase(parameterizedtestcase.ParameterizedTestCase):
+    # compression | ignore_ext | behavior |
+    # ----------- | ---------- | -------- |
+    # 'gz'        | False      | Override |
+    # 'bz2'       | False      | Override |
+    @parameterizedtestcase.ParameterizedTestCase.parameterize(
+        ("_compression", "decompressor"),
+        [
+            ("gz", gzip.decompress),
+            ("bz2", bz2.decompress),
+        ],
+    )
+    @mock_s3
+    def test_rw_compression_prescribed(self, _compression, decompressor):
+        """Should read/write files with `_compression`, as prescribed."""
+        s3 = boto3.resource("s3")
+        s3.create_bucket(Bucket="bucket").wait_until_exists()
+        key = "s3://bucket/key.txt"
+
+        with smart_open.open(key, "wb", compression=_compression) as fout:
+            fout.write(_RAW_DATA)
+
+        #
+        # Check that what we've created is compressed as expected.
+        #
+        with smart_open.open(key, "rb", compression=NO_COMPRESSION) as fin:
+            data = decompressor(fin.read())
+            assert data == _RAW_DATA
+
+        #
+        # We should be able to read it back as well.
+        #
+        with smart_open.open(key, "rb", compression=_compression) as fin:
+            assert fin.read() == _RAW_DATA
+
+    # compression | ignore_ext | behavior |
+    # ----------- | ---------- | -------- |
+    # 'extension' | False      | Enable   |
+    # 'none'      | False      | Disable  |
+    @parameterizedtestcase.ParameterizedTestCase.parameterize(
+        ("_compression", "decompressor", "compression_kwargs", "no_compression_kwargs"),
+        [
+            (
+                "gz",
+                gzip.decompress,
+                dict(compression=INFER_FROM_EXTENSION),
```

Comment:
I think you're misusing parameterized tests here. The compression_kwargs and no_compression_kwargs are identical across the two tests, so there is no point in parameterizing them.

Reply:
Totally fair. This simplification fell out of previous refactoring and I missed it. Will fix.

- [ ] Remove constant parameters